internal/bminventory/inventory.go

```diff
@@ -2062,6 +2066,15 @@ func (b *bareMetalInventory) updateNetworkParams(params installer.UpdateClusterP
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
 
+	if err = validations.ValidateNetworkTypeWithIPv6(params.ClusterUpdateParams.NetworkType, machineCidr); err != nil {
+		validNetworkTypes := []string{models.ClusterNetworkTypeOVNKubernetes, models.ClusterNetworkTypeOpenShiftSDN}
+		if !funk.ContainsString(validNetworkTypes, swag.StringValue(params.ClusterUpdateParams.NetworkType)) {
```

Comment:
The swagger defines an enum, doesn't the code generated from the swagger check this?

Reply:
Is this also enforced in non-REST API?

Reply:
@ori-amizur can you please explain what you mean by non-REST API?

Reply:
Hive ?